pallets/attestation/src/lib.rs

```diff
@@ -348,9 +357,105 @@ pub mod pallet {
 				},
 			);
 
-			Self::deposit_event(Event::AttestationRevoked(revoker, claim_hash));
+			Self::deposit_event(Event::AttestationRevoked(who, claim_hash));
+
+			Ok(Some(<T as pallet::Config>::WeightInfo::revoke(delegation_depth)).into())
+		}
+
+		/// Remove an attestation.
+		///
+		/// The origin must be either the creator of the attestation or an
+		/// entity which is an ancestor of the attester in the delegation tree,
+		/// i.e., it was either the delegator of the attester or an ancestor
+		/// thereof.
+		///
+		/// Emits `AttestationRemoved`.
+		///
+		/// # <weight>
+		/// Weight: O(P) where P is the number of steps required to verify that
+		/// the dispatch Origin controls the delegation entitled to revoke the
+		/// attestation. It is bounded by `max_parent_checks`.
+		/// - Reads: [Origin Account], Attestations, delegation::Roots
+		/// - Reads per delegation step P: delegation::Delegations
+		/// - Writes: Attestations, DelegatedAttestations
+		/// # </weight>
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::remove(*max_parent_checks))]
+		pub fn remove(
+			origin: OriginFor<T>,
+			claim_hash: ClaimHashOf<T>,
+			max_parent_checks: u32,
+		) -> DispatchResultWithPostInfo {
+			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
+			let who = source.subject();
+
+			let attestation = Attestations::<T>::get(&claim_hash).ok_or(Error::<T>::AttestationNotFound)?;
+
+			let delegation_depth = if attestation.attester != who {
+				Self::verify_delegated_access(&who, &attestation, max_parent_checks)?
+			} else {
+				0
+			};
+
+			// *** No Fail beyond this point ***
+
+			log::debug!("removing Attestation");
+			Pallet::<T>::free_deposit(&attestation.deposit);
+			Attestations::<T>::remove(&claim_hash);
+			if let Some(delegation_id) = attestation.delegation_id {
+				DelegatedAttestations::<T>::mutate(&delegation_id, |maybe_attestation| {
```

Comment:
W: I would call it `maybe_attestations` or simple `attestations`.

Reply:
cbfbb55f9b162679b05206e8e1a45af25d30993c